model/src/main/kotlin/Identifier.kt

```diff
@@ -78,6 +78,14 @@ data class Identifier(
         }
     }
 
+    init {
+        require(!provider.contains(":") && !namespace.contains(":") && !name.contains(":") && !version.contains(":")) {
```

Comment:
Maybe nicer:

    listOf(provider, namespace, name, version).none { it.contains(":") }

Reply:
Done.